src/runtime-rs/crates/shim/src/panic_hook.rs

```diff
@@ -36,6 +36,20 @@ pub(crate) fn set_panic_hook() {
             "A panic occurred at {}:{}: {}\r\n{:?}", filename, line, cause, bt_data
         );
 
+        // print panic log to dmesg
+        // The panic log size is too large to /dev/kmsg, so write by line.
+        if let Ok(mut file) = OpenOptions::new().write(true).open("/dev/kmsg") {
```

Comment:
Again, it would be better to define this path outside of the function to allow it to be unit tested. Possibly outside of the scope of this PR, but I'd also consider making the panic hook an actual function rather than a closure to allow that to be tested too (because if the system panics, we *really* want this code to work as expected ;)

Reply:
Thanks for your review, change to use a global variable.